radix-engine/src/kernel/kernel.rs

```diff
@@ -50,30 +40,27 @@ pub struct Kernel<
     /// Heap
     pub(super) heap: Heap,
     /// Store
-    pub(super) track: &'g mut Track<'s, R>,
+    pub(super) track: &'g mut Track<'s>,
 
     /// ID allocator
     pub(super) id_allocator: &'g mut IdAllocator,
     /// Interpreter capable of running scrypto programs
     pub(super) scrypto_interpreter: &'g ScryptoInterpreter<W>,
-    /// Kernel module
-    pub(super) module: &'g mut M,
+    /// Kernel module mixer
+    pub(super) module: KernelModuleMixer,
```

Comment:
Didn't want to use generics for this?

Reply:
I tried this but ran into some difficulties with `get_module_state`. 

Essentially, it requires me to implement a generic module storage (which isn't useful in this stage).

Reply:
Sounds good.